### PR TITLE
Expose ability to set OptimizationProtocols for for optimizations, torsiondrives

### PR DESCRIPTION
**Pull request by dotsdl**
## Description
[`OptimizationProtocols`](https://github.com/MolSSI/QCElemental/blob/1ef6b52adc4a18cfd6a2cfc1b56880daaeea3a44/qcelemental/models/procedures.py#L32-L48) allow users to e.g. only save gradients, wavefunctions for starting and final geometries of an optimization, reducing storage requirements for results. The changes here expose this functionality through QCSubmit, allowing users to set `OptimizationProtocols` settings other than the defaults if desired.

## Todos
  - [x] add tests

## Status
- [x] Ready to go

### CODE REVIEW

File: openff/qcsubmit/datasets/datasets.py [hunk 1]
```diff
@@ -907,6 +907,10 @@ class OptimizationDataset(BasicDataset):
         GeometricProcedure(),
         description="The optimization program and settings that should be used.",
     )
+    protocols: OptimizationProtocols = Field(
+        OptimizationProtocols(),
+        description="Protocols regarding the manipulation of a Optimization output data.",
```
Comment (lilyminium) [inline openff/qcsubmit/datasets/datasets.py:912]:
```suggestion
        description="Protocols regarding the manipulation of Optimization output data.",
```

Comment (dotsdl) [reply openff/qcsubmit/datasets/datasets.py:912]: Thanks! Got it!

File: openff/qcsubmit/tests/test_datasets.py [hunk 2]
```diff
@@ -2610,6 +2611,25 @@ def test_optimizationdataset_qc_spec():
...
+        description="XXXXXXXX",
+    )
+
+    assert dataset.protocols.trajectory == "all"
+
+    # change the protocol to only save gradient evaluations for final state
+    dataset.protocols = OptimizationProtocols(trajectory="final")
+
+    assert dataset.protocols.trajectory == "final"
+
```
Comment (lilyminium) [inline openff/qcsubmit/tests/test_datasets.py:2631]: Would it be worth also adding a test to `test_submissions` that actually carries out the calculation and checks that only the final gradients are saved, or would we expect that to be covered by the QC* stack so long as our inputs have the keyword?

Comment (dotsdl) [reply openff/qcsubmit/tests/test_datasets.py:2631]: I think it makes sense to do this. I'll modify the existing `test_optimization_submissions` to use `'final'` for the `trajectory` protocol and see if this gives the expected result.

Comment (dotsdl) [reply openff/qcsubmit/tests/test_datasets.py:2631]: Added!

**Review by lilyminium — APPROVED** [review]
Thanks for adding the protocol and test @dotsdl -- everything LGTM pending green CI!

**Comment by dotsdl** [comment]
Thanks @lilyminium!

**Pull request merged**